source/features/quick-label-removal.tsx

```diff
@@ -50,17 +50,15 @@ function addRemoveLabelButton(label: HTMLElement): void {
 	);
 }
 
-async function init(): Promise<Deinit> {
+async function init(signal: AbortSignal): Promise<Deinit> {
 	await api.expectToken();
 
-	return [
-		observe('.js-issue-labels .IssueLabel:not(.rgh-quick-label-removal-already-added)', {
-			constructor: HTMLElement,
-			add: addRemoveLabelButton,
-		}),
+	delegate(document, '.rgh-quick-label-removal:not([disabled])', 'click', removeLabelButtonClickHandler, {signal});
 
-		delegate(document, '.rgh-quick-label-removal:not([disabled])', 'click', removeLabelButtonClickHandler),
-	];
+	return observe('.js-issue-labels .IssueLabel:not(.rgh-quick-label-removal-already-added)', {
+		constructor: HTMLElement,
+		add: addRemoveLabelButton,
+	});
```

Comment:
I propose wrapping `observe` to enable:

- signal support
- selector parsing for the types
- array of selectors
- simpler listener (`observe(selector, callback)` instead of `observe(selector, {add: callback})`)

and potentially, separately:
- automatic class addition via `getSnapshotUUID`: if we have CSS relying on certain classes we can still add those manually, on top of the automatic class

Can be done in 1-2 additional PRs or added to #5222